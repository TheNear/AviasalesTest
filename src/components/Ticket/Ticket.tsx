import React from "react";
import TicketInfo from "../../componentsHelper/TicketInfo/TicketInfo";
import style from "./Ticket.module.css";

const Ticket: React.FC = () => {
  return (
    <li className={style.ticket}>
      <div className={style.price}>13 400 P</div>
      <img
        className={style.image}
        src="http://pics.avs.io/99/36/S7.png"
        alt="airport"
      />
      <TicketInfo />
      <TicketInfo />
      <TicketInfo />
      <TicketInfo />
      <TicketInfo />
      <TicketInfo />
    </li>
  );
};

export default Ticket;
