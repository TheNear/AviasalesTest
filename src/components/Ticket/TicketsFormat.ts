export const stopsFormat: Array<string> = [
  "Без пересадок",
  "1 Пересадка",
  "2 Пересадки",
  "3 Пересадки",
  "4 Пересадки",
  "5 Пересадок",
  "6 Пересадок",
  "7 Пересадок",
  "8 Пересадок",
  "9 Пересадок",
  "10 Пересадок",
];

export const convertPrice = (number: number): string => {
  return new Intl.NumberFormat("ru-RU").format(number);
};

export const getTravelTime = (time: number): string => {
  const dateObj = new Date(time * 1000 * 60);
  const hours = dateObj.getHours();
  const minutes = dateObj.getMinutes();
  return `${hours}ч ${minutes}м`;
}