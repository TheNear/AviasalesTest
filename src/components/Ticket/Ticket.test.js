import React from "react";
import { shallow } from "enzyme";
import { Ticket } from "./Ticket";
import { ticketMock } from "./ticketsMock";

describe("Test Ticket component", () => {
  it("Render empty Ticket", () => {
    const wrapper = shallow(<Ticket ticket={ticketMock} />);
    expect(wrapper).toMatchSnapshot();
  });
});
