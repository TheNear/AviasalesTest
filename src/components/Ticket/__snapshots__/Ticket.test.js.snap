// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Ticket component Render empty Ticket 1`] = `
<li
  className="ticket box"
>
  <div
    className="price"
  >
    18,204
     P
  </div>
  <img
    alt="airport"
    className="image"
    src="http://pics.avs.io/99/36/S7.png"
  />
  <TicketInfo
    text="16:46 - 18:30"
    title="HKT - HKT"
  />
  <TicketInfo
    text="25ч 44м"
    title="В пути"
  />
  <TicketInfo
    text="SHA,DXB,HKG"
    title="3 Пересадки"
  />
  <TicketInfo
    text="16:46 - 01:10"
    title="HKT - HKT"
  />
  <TicketInfo
    text="32ч 24м"
    title="В пути"
  />
  <TicketInfo
    text="SHA,DXB,HKG"
    title="3 Пересадки"
  />
</li>
`;
